improver/utilities/cube_checker.py

```diff
@@ -203,3 +203,41 @@ def spatial_coords_match(cubes: Union[List, CubeList]) -> bool:
             and match
         )
     return match
+
+
+def assert_time_coords_valid(inputs: List[Cube], time_bounds: bool):
+    """
+    Raises appropriate ValueError if
+
+    - Any input cube has or is missing time bounds (depending on time_bounds)
+    - Input cube times and forecast_reference_times do not match
+
+    Args:
+        inputs:
+            List of Cubes where times should match
+        time_bounds:
+            When True, time-bounds are checked for and compared on the input cubes.
+            When False, the absence of time-bounds is checked for.
```

Comment:
Is is correct that you don't intend to support these tests being disabled i.e. you always intend to either check the time bounds or check the absence time bounds? I can see why this interface is like it is, but it's also slightly surprising that `False` doesn't disable the checking of the time bounds, which would have been my assumption.

Reply:
Yes, this is my intention. Either time bounds should be present and matching on all cubes, or they should be absent on all cubes. I cannot think of a case where you don't care whether time bounds are present, matching or otherwise.